examples/scanner/scanner.ino

```diff
@@ -38,74 +60,100 @@ RF24 radio(CE_PIN, CSN_PIN);
+bool constCarrierMode = 0;  // this flag controls example behavior (scan mode is default)
+
+void printHeader();  // prototype function for printing the channels' header
 
-//
-// Setup
-//
 
 void setup(void) {
-  //
-  // Print preamble
-  //
 
+  // Print preamble
   Serial.begin(115200);
-  printf_begin();
-  Serial.println(F("\n\rRF24/examples/scanner/"));
+  while (!Serial) {
+    // some boards need this to wait for Serial connection
+  }
+  Serial.println(F("RF24/examples/scanner/"));
+  Serial.println(F("!!! Please do not send line endings in the serial monitor !!!"));
 
-  //
   // Setup and configure rf radio
-  //
+  if (!radio.begin()) {
+    Serial.println(F("radio hardware not responding!"));
+    while (true) {
+      // hold in an infinite loop
+    }
+  }
+  radio.setAutoAck(false);   // Don't acknowledge arbitrary signals
+  radio.disableCRC();        // Accept any signal we find
+  radio.setAddressWidth(2);  // A reverse engineering tactic (not typically recommended)
+  for (uint8_t i = 0; i < 6; ++i) {
+    radio.openReadingPipe(i, noiseAddress[i]);
+  }
 
-  radio.begin();
-  radio.setAutoAck(false);
+  // set the data rate
+  Serial.print(F("Select your Data Rate. "));
+  Serial.print(F("Enter '1' for 1 Mbps, '2' for 2 Mbps, '3' for 250 kbps. "));
+  Serial.println(F("Defaults to 1Mbps."));
+  while (!Serial.available()) {
+    // wait for user input
+  }
+  uint8_t dataRate = Serial.parseInt();
+  if (dataRate == 50) {
+    Serial.println(F("Using 2 Mbps."));
+    radio.setDataRate(RF24_2MBPS);
+  } else if (dataRate == 51) {
+    Serial.println(F("Using 250 kbps."));
+    radio.setDataRate(RF24_250KBPS);
+  } else {
+    Serial.println(F("Using 1 Mbps."));
+    radio.setDataRate(RF24_1MBPS);
+  }
+  Serial.println(F("***Enter a channel number to emit a constant carrier wave."));
+  Serial.println(F("***Enter a negative number to switch back to scanner mode."));
 
   // Get into standby mode
   radio.startListening();
   radio.stopListening();
-  radio.printDetails();
-
-  //delay(1000);
-  // Print out header, high then low digit
-  int i = 0;
-  while (i < num_channels) {
-    Serial.print(i >> 4, HEX);
-    ++i;
-  }
-  Serial.println();
-  i = 0;
-  while (i < num_channels) {
-    Serial.print(i & 0xf, HEX);
-    ++i;
-  }
-  Serial.println();
-  //delay(1000);
-}
+  radio.flush_rx();
 
-//
-// Loop
-//
+  // printf_begin();
+  // radio.printPrettyDetails();
+  // delay(1000);
 
-const int num_reps = 100;
-bool constCarrierMode = 0;
+  // Print out vertical header
+  printHeader();
+}
 
 void loop(void) {
   /****************************************/
-  // Send g over Serial to begin CCW output
-  // Configure the channel and power level below
+  // Send a number over Serial to begin Constant Carrier Wave output
+  // Configure the power amplitude level below
   if (Serial.available()) {
-    char c = Serial.read();
-    if (c == 'g') {
+    int8_t c = Serial.parseInt();
```

Comment:
I added a `while (Serial.peek() != -1)` after this block to discard line endings. Otherwise, it would interpret line endings as erroneous user input, clamp the channel number to 0 and begin a CCW. That shouldn't happen now that `while (Serial.peek() != -1)` discards the CRLF and LF chars.